src/kernel/src/ucred/prison.rs

```diff
@@ -0,0 +1,80 @@
+use bitflags::bitflags;
+use std::borrow::Cow;
+
+#[derive(Debug)]
+pub struct Prison {
+    id: u32, // An internal id to use for comparisons;
+    parent: Option<Cow<'static, Self>>,
+    flags: PrisonFlags,
+    allow: PrisonAllow,
+}
+
+impl Prison {
+    pub fn flags(&self) -> PrisonFlags {
+        self.flags
+    }
+
+    pub fn allow(&self) -> PrisonAllow {
+        self.allow
+    }
+
+    pub fn is_child(&self, other: &Self) -> bool {
+        let mut p = other.parent.as_ref();
+
+        while let Some(pr) = p {
+            if self == other {
```

Comment:
I think this one is not correct.

Reply:
<img width="412" alt="image" src="https://github.com/obhq/obliteration/assets/48620541/605ba2c9-e2b1-46c3-8d92-aaa2d89257ca">

Reply:
What you did here is always checking if `self` is equal to `other` every loop.

Reply:
OH :skull: